### ISSUE TITLE
AssertionError when calling auto_backup services

### ISSUE THREAD

**Issue by Apocrathia** [jcwillox/hass-auto-backup#19]
I recently noticed that my backups haven't run in almost 2 months. [Everything is controlled via automations](https://github.com/Apocrathia/home-assistant-config/blob/master/packages/system/backup.yaml)

While investigating, I tried triggering an automation manually, and I'm getting the following in my log:

```
ERROR (MainThread) [homeassistant] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/config/custom_components/auto_backup/__init__.py", line 306, in new_snapshot
    self.update_sensor_callback()
  File "/config/custom_components/auto_backup/sensor.py", line 32, in update_callback
    self.async_schedule_update_ha_state(True)
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 423, in async_schedule_update_ha_state
    assert self.hass is not None
AssertionError
```

I've tried uninstalling and reinstalling the integration from HACS, but I'm still getting the same error. Is there something that I'm missing?

**Comment by jcwillox** [comment]
Wow, that's quite a while, you might want to set up [failure notifications](https://github.com/jcwillox/hass-auto-backup#example-notify-on-snapshot-failure).
It seems it's an issue with the sensor Auto Backup creates `sensor.auto_backup`, did you delete this entity from the UI?

**Comment by Apocrathia** [comment]
[Already have, and I never got notified.](https://github.com/Apocrathia/home-assistant-config/blob/master/packages/system/backup.yaml#L75)

The `sensor.auto_backup` entity is still there, and hasn't been deleted. 
![image](https://user-images.githubusercontent.com/1345711/91398974-56a95680-e7fa-11ea-8f39-ad786f58fc6b.png)
![image](https://user-images.githubusercontent.com/1345711/91399057-86585e80-e7fa-11ea-83c8-ee9170bc2978.png)

**Comment by jcwillox** [comment]
Oh right I'll definitely look into that, clearly I need better exception handling because an error when updating the sensor should not be affecting the backup/failure notifications.

**Comment by Apocrathia** [comment]
Let me know if there's any logs I can provide, or debug lines I can add to get some more feedback.

**Comment by jcwillox** [comment]
Thanks, I've looked at the code and I can't find any reason that `self.hass is None` for the sensor. I'm going to add a broad try/catch around the update method, this won't stop the error from occurring but will mean that the backups should continue to work.
For the time being you could try disabling the sensor and restarting.
Also, what version of Home Assistant are you running? and do you know if this was the same error you were getting over the past few months?

**Comment by Apocrathia** [comment]
I tried doing another uninstall/reinstall via HACS with reboots in between, but this this time I also removed the `.storage/auto_backup.snapshots_expiry` file. Still getting the same error. Would there be anything else cached that I should try and clear out?

**Comment by jcwillox** [comment]
I've just released [`0.6.0`](https://github.com/jcwillox/hass-auto-backup/releases/tag/0.6.0), this decouples the sensor from the snapshot method, which should stop that error from occurring, and hopefully, your backups should start working.

**Comment by Apocrathia** [comment]
Yep. That fixed it. Everything is working now!
